src/registry/softRequiredErrors/edit.tsx

```diff
@@ -0,0 +1,122 @@
+import {JSONEditor} from '@open-formulieren/monaco-json-editor';
+import {SoftRequiredErrorsComponentSchema} from '@open-formulieren/types';
+import {useFormikContext} from 'formik';
+import React, {useContext, useEffect, useRef, useState} from 'react';
+
+import PreviewModeToggle, {PreviewState} from '@/components/PreviewModeToggle';
+import {Key, RichText} from '@/components/builder';
+import {Tab, TabList, TabPanel, Tabs} from '@/components/formio';
+import {BuilderContext} from '@/context';
+import {useErrorChecker} from '@/utils/errors';
+
+import {EditFormDefinition} from '../types';
+
+/**
+ * Form to configure a Formio 'content' type component.
+ *
+ * W/r to translations, the 'NL' language is considered the default, and the main html
+ * value is populated from that field (TODO: implement this).
+ */
+const EditForm: EditFormDefinition<SoftRequiredErrorsComponentSchema> = () => {
+  const {uniquifyKey, supportedLanguageCodes, theme} = useContext(BuilderContext);
+  const isKeyManuallySetRef = useRef(false);
+  const {values, setFieldValue, setValues} = useFormikContext<SoftRequiredErrorsComponentSchema>();
+  const generatedKey = uniquifyKey(values.key);
+
+  const [previewMode, setPreviewMode] = useState<PreviewState>('rich');
+
+  const defaultLanguageCode = supportedLanguageCodes[0] ?? 'nl';
+
+  // Synchronize the default/first language tab value to the main `html` field.
+  useEffect(() => {
+    const currentValue = values.openForms?.translations?.[defaultLanguageCode]?.html;
+    if (currentValue === undefined && values.html) {
+      // if we have a 'html' value, but no default translation -> store the default translation
+      setFieldValue(`openForms.translations.${defaultLanguageCode}.html`, values.html);
+    } else if (values.html !== currentValue) {
+      // otherwise sync the value of the translation field to the main field.
+      setFieldValue('html', currentValue);
+    }
+  });
```

Comment:
I'm not entirely sure what the consequences are, but shouldn't we define the dependencies to make sure this useEffect is only triggered when the `values` or the `setFieldValue` change?

Reply:
this function just executes on every render (it's copied from the `content` component), so at worst the hook runs a couple times too often.

Normally I'd indeed optimize this by specifying dependencies, but `values` has a stable identity (I think) from Formik, and if it didn't, then the hook would run every time anyway. We have the proper conditions that check if an update is needed or not, so you don't get infinite loops :)